### PR TITLE
ref: Move resolve function and implement support for local OpenAPI paths

### PR DESCRIPTION
**Pull request by NisanthanNanthakumar**
## Objective
We want to work locally with API docs so that people can make changes to the OpenAPI spec in Sentry but also be able to see what it would look like in docs.sentry.io/api/.

In this PR, I have implemented support for developers to set an environment variable `OPENAPI_LOCAL_PATH` that can be lead to their locally generated `openapi-derefed.json`. 

If it is able to resolve the path, `gatsby-plugin-openapi` will generate its' graphql nodes from the local `openapi-derefed.json`, else it defaults to fetching the `openapi-derefed.json` from the versioned commit in `getsentry/sentry-api-schema`.

ex: `OPENAPI_LOCAL_PATH=/Users/nisanthan.nanthankumar/Documents/sentry-api-schema/openapi-derefed.json yarn start`

## Future Improvements (Not this PR)
 - Support Hot-Reload. Maybe we can watch the file from the env variable. **Need to look into this**

### CODE REVIEW

**Review by ceorourke — COMMENTED** [review]
This is great! I just tried it out with my path to `sentry-api-schema` and then in `sentry`. Hot reload would definitely be a cool future addition. I doubt you want me reviewing the JS though.

**Comment by MimiDumpling** [comment]
Thanks for updating the API docs! Gonna leave the technical review to the Eco team. 🌻

**Comment by NisanthanNanthakumar** [comment]
> This is great! I just tried it out with my path to `sentry-api-schema` and then in `sentry`. Hot reload would definitely be a cool future addition. I doubt you want me reviewing the JS though.

@ceorourke Haha awesome! But you should definitely review the JS!!

File: src/gatsby/utils/resolveOpenAPI.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      console.log(`Fetching from ${process.env.OPENAPI_LOCAL_PATH}`);
+      let data = await fs.readFile(process.env.OPENAPI_LOCAL_PATH, "utf8");
+      return data;
+    } catch (error) {
+      console.log(
+        `Failed to connect to  ${process.env.OPENAPI_LOCAL_PATH}. Continuing to fetch versioned schema from Github.
+        ${error}`
+      );
+    }
+  }
```
Comment (MeredithAnya) [inline src/gatsby/utils/resolveOpenAPI.ts:19]: I think if you wanted you could add another `console.log` indicating that you are fetching from the `sentry-api-schema`. I forgot to put `export` in front of the `OPENAPI_LOCAL_PATH` and was confused for a min about why I didn't see the console.log message.

**Review by MeredithAnya — APPROVED** [review]
Had one small comment. But looks good. 

I think that we shouldn't really be using the local `sentry-api-schema` repo unless we are saying that the flow to changing and testing out schema changes is to have the sentry command `build-derefed-docs` output the derefed to your local version of that repo, right? But seems like people shouldn't have to have 3 repos to test out the docs. 

So I think the next step would be to figure out where does the local openapi-derefed.json go.

File: src/gatsby/utils/resolveOpenAPI.ts (hunk 1 above)
Comment (NisanthanNanthakumar) [reply src/gatsby/utils/resolveOpenAPI.ts:19]: @MeredithAnya hmm should we log defaults? I don't think we currently do that for anything else.

**Comment by NisanthanNanthakumar** [comment]
> Had one small comment. But looks good.
> 
> I think that we shouldn't really be using the local `sentry-api-schema` repo unless we are saying that the flow to changing and testing out schema changes is to have the sentry command `build-derefed-docs` output the derefed to your local version of that repo, right? But seems like people shouldn't have to have 3 repos to test out the docs.
> 
> So I think the next step would be to figure out where does the local openapi-derefed.json go.

I used my `sentry-api-schema` as an example. I expect people to connect to their `sentry` derefed. But its generalized to be flexible for any use case.

**Comment by manuzope** [comment]
@NisanthanNanthakumar can we update the reference here: https://develop.sentry.dev/docs/api/#how-to-make-an-endpoint-public.

Once we're done with developer experience improvements, might also be a good idea to add a blurb on that page.

**Pull request merged**